File: 1732.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        int curr = 0, highest =0;
        for(int i : gain){
            curr += i;
            highest = max(highest, curr);
        }
        return highest;
    }
};

int main(){
    Solution s;
    int n;
    cin >> n;
    vector<int> gain(n);
    cout << s.largestAltitude(gain) << endl;
    return 0;
}

File: python/1732.py
class Solution:
    def largestAltitude(self, gain: list[int]) -> int:
        curr, highest = 0, 0
        for i in gain:
            curr += i
            highest = max(highest, curr)
        return highest
    
if __name__ == '__main__':
    gain = list(map(int, input().split()))
    print(Solution().largestAltitude(gain))